src/scripts/librarymenu.js

```diff
@@ -155,297 +302,671 @@ define(["dom", "layoutManager", "inputManager", "connectionManager", "events", "
+                    sidebarLinks[i].addEventListener("click", onSidebarLinkClick);
+                }
+            });
+        }
     }
 
     function getTopParentId() {
-        return getParameterByName("topParentId") || null
+        return getParameterByName("topParentId") || null;
     }
 
     function onMainDrawerClick(e) {
-        dom.parentWithTag(e.target, "A") && setTimeout(closeMainDrawer, 30)
+        if (dom.parentWithTag(e.target, "A")) {
+            setTimeout(closeMainDrawer, 30);
+        }
     }
 
     function onLogoutClick() {
-        Dashboard.logout()
+        Dashboard.logout();
     }
 
     function updateCastIcon() {
-        var context = document,
-            info = playbackManager.getPlayerInfo(),
-            icon = headerCastButton.querySelector("i");
-        info && !info.isLocalPlayer ? (icon.innerHTML = "&#xE308;", headerCastButton.classList.add("castButton-active"), context.querySelector(".headerSelectedPlayer").innerHTML = info.deviceName || info.name) : (icon.innerHTML = "&#xE307;", headerCastButton.classList.remove("castButton-active"), context.querySelector(".headerSelectedPlayer").innerHTML = "")
+        var context = document;
+        var info = playbackManager.getPlayerInfo();
+        var icon = headerCastButton.querySelector("i");
+
+        if (info && !info.isLocalPlayer) {
+            icon.innerHTML = "&#xE308;";
+            headerCastButton.classList.add("castButton-active");
+            context.querySelector(".headerSelectedPlayer").innerHTML = info.deviceName || info.name;
+        } else {
+            icon.innerHTML = "&#xE307;";
+            headerCastButton.classList.remove("castButton-active");
+            context.querySelector(".headerSelectedPlayer").innerHTML = "";
+        }
     }
 
     function updateLibraryNavLinks(page) {
-        var i, length, isLiveTvPage = page.classList.contains("liveTvPage"),
-            isChannelsPage = page.classList.contains("channelsPage"),
-            isEditorPage = page.classList.contains("metadataEditorPage"),
-            isMySyncPage = page.classList.contains("mySyncPage"),
-            id = isLiveTvPage || isChannelsPage || isEditorPage || isMySyncPage || page.classList.contains("allLibraryPage") ? "" : getTopParentId() || "",
-            elems = document.getElementsByClassName("lnkMediaFolder");
+        var i;
+        var length;
+        var isLiveTvPage = page.classList.contains("liveTvPage");
+        var isChannelsPage = page.classList.contains("channelsPage");
+        var isEditorPage = page.classList.contains("metadataEditorPage");
+        var isMySyncPage = page.classList.contains("mySyncPage");
+        var id = isLiveTvPage || isChannelsPage || isEditorPage || isMySyncPage || page.classList.contains("allLibraryPage") ? "" : getTopParentId() || "";
+        var elems = document.getElementsByClassName("lnkMediaFolder");
+
         for (i = 0, length = elems.length; i < length; i++) {
```

Comment:
Should just move the variable declaration in here.

Reply:
What do you mean?

Reply:
`for (var i = 0; i < elems.length; i++) {` and then remove line 638 and 639.